### DISCUSSION TITLE
更新2.3.0版本以后 重启程序一直卡在正在启动服务……

### DISCUSSION THREAD

**Discussion by Michael121318** [Chia-Network/chia-blockchain#18034] [Support]
更新2.3.0版本以后 重启程序一直卡在正在启动服务……
![微信截图_20240516234532](https://github.com/Chia-Network/chia-blockchain/assets/147701612/463e1954-0bbb-4b26-a294-f5a6d8920c60)

**Comment by Michael121318** [comment]
这个怎么解决呢

**Comment by BrandtH22** [comment]
Hey @Michael121318 , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia). 

With the 2.2.x+ versions the blockchain database needs to be of version 2. There are two methods to get the version 2 database based on the amount of storage you have available on your device. If you have at least 300GB of free space then I recommend using the CLI command, if not then use the torrent download option:

Steps for the CLI command:
```
1. Stop the chia client (this can be done while chia is running but stopping the client saves from any potential issues)
2. Run the command `chia db upgrade` (you do not need the input or output paths if these are still default)
- This process will take some time and requires additional space as the v1 db is compacted into the v2 db which leaves the v1 db intact
3. Verify the config file (~\.chia\mainnet\config\config.yaml) has the correct value under the full_node section for `database_path: db/blockchain_v2_CHALLENGE.sqlite` (should only need to change the v1 to v2)
4. After the above completes start the chia client
5. If all loads up properly (might take upwards of 30 minutes to load) then you can safely delete the v1 db found here (~\.chia\mainnet\db\blockchain_v1_mainnet.sqlite)
```

Steps for using the torrent: 
```
1. Download the torrent file from https://www.chia.net/downloads/#database-checkpoint
2. Use a torrent client to download the full db
3. Unpack/reassemble the torrent file that was downloaded (on windows I used 7zip, Mac and linux have built in tools that work for this)
4. Move the db to the correct folder (~\.chia\mainnet\db\) and update the name to "blockchain_v2_mainnet.sqlite" (removing the date information in the name)
5. Verify the config file (~\.chia\mainnet\config\config.yaml) has the correct value under the full_node section for `database_path: db/blockchain_v2_CHALLENGE.sqlite` (should only need to change the v1 to v2)
6. Launch chia and wait for a bit (the height to hash and peers files need to be built so this can take upwards of 30 minutes)
```
---
The following has been translated using translation software and may contain errors
以下内容已使用翻译软件翻译，可能包含错误
---

嘿@Michael121318，作为提醒，我们通常可以在我们的 discord 服务器 （https://discord.gg/chia） 中提供更及时和彻底的支持。

对于 2.2.x+ 版本，区块链数据库需要是版本 2。有两种方法可以根据设备上的可用存储量获取版本 2 数据库。如果您至少有 300GB 的可用空间，那么我建议使用 CLI 命令，如果没有，则使用 torrent 下载选项：

CLI 命令的步骤： 
```
1. 停止 chia 客户端（这可以在 chia 运行时完成，但停止客户端保存不会出现任何潜在问题） 
2. 运行命令"chia db upgrade"（如果输入或输出路径仍为默认路径，则不需要这些路径） - 此过程将花费一些时间，并且需要额外的空间，因为 v1 db 被压缩到 v2 db 中，从而保持 v1 db 不变 
3. 验证配置文件 （~\.chia\mainnet\config\config.yaml） 在"database_path： db/blockchain_v2_CHALLENGE.sqlite"的 full_node 部分下是否具有正确的值（只需将 v1 更改为 v2） 
4. 完成上述操作后，启动 chia 客户端 
5. 如果所有加载都正确（加载可能需要 30 分钟以上），那么您可以安全地删除此处找到的 v1 数据库 （~\.chia\mainnet\db\blockchain_v1_mainnet.sqlite）
```

使用种子的步骤： 
```
1.从 https://www.chia.net/downloads/#database-checkpoint 下载torrent文件 
2. 使用 torrent 客户端下载完整的数据库 
3. 解压缩/重新组装下载的 torrent 文件（在 Windows 上我使用 7zip，Mac 和 linux 内置了可用于此的工具） 
4. 将数据库移动到正确的文件夹 （~\.chia\mainnet\db\） 并将名称更新为"blockchain_v2_mainnet.sqlite"（删除名称中的日期信息） 
5. 验证配置文件 （~\.chia\mainnet\config\config.yaml） 在"database_path： db/blockchain_v2_CHALLENGE.sqlite"的 full_node 部分下是否具有正确的值（只需将 v1 更改为 v2） 
6. 启动 chia 并稍等片刻（需要构建哈希和对等文件的高度，因此这可能需要 30 分钟以上）
```

**Reply by Michael121318** [reply]
After downloading the torrent file again and following the steps, the program still cannot be started

**Reply by BrandtH22** [reply]
Hey @Michael121318 , as a reminder we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

Thank you for providing the logs in the issue ticket, that shed light on the issue and the solution has been provided on that issue ticket.

Please review the information that William provided here:
https://github.com/Chia-Network/chia-blockchain/issues/18035#issuecomment-2137838846

Do let us know if any questions arise. Thank you and make it an excellent day,
Brandt H

**Comment by BrandtH22** [comment]
Hey @Michael121318 , as a reminder we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

Thank you for providing the logs in the issue ticket, that shed light on the issue and the solution has been provided on that issue ticket.

Please review the information that William provided here:
https://github.com/Chia-Network/chia-blockchain/issues/18035#issuecomment-2137838846

Do let us know if any questions arise. Thank you and make it an excellent day,
Brandt H
